src/store/slices/testingHistory.slice.ts

```diff
@@ -0,0 +1,106 @@
+import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
+import { fetch } from "api";
+
+import { processFinishedJson } from "compositions/Timeline/components/TimelineItem/timeline.helper";
+import { isAnyTaskFailure } from "pages/certification/Certification.helper";
+
+import type { Run } from 'components/CreateCertificate/CreateCertificate';
+
+interface Certificate {
+  transactionId: string;
+}
+
+interface TestingHistoryState {
+  loading: boolean;
+  history: Run[];
+  certificates: Map<string, Certificate>;
+}
+
+const initialState: TestingHistoryState = {
+  loading: false,
+  history: [],
+  certificates: new Map(),
+};
+
+export const fetchHistory = createAsyncThunk("fetchHistory", async (payload: {}, thunkApi) => {
+  try {
+    const response = await fetch<Run[]>(thunkApi, { method: 'GET', url: '/run' });
+    return response.data;
+  } catch (e: any) {
+    return thunkApi.rejectWithValue(e.response.data);
+  }
+});
+
+export const updateRowStatus = createAsyncThunk("updateRowStatus", async (payload: { index: number, runId: string }, thunkApi) => {
```

Comment:
`updateRowStatus` could be renamed to `getRowStatus`, since it is not an update happening here

Reply:
Done!